sdk/openai/Azure.AI.OpenAI.Assistants/src/Generated/UploadFileRequest.Serialization.cs

```diff
@@ -27,7 +28,14 @@ void IJsonModel<UploadFileRequest>.Write(Utf8JsonWriter writer, ModelReaderWrite
 
             writer.WriteStartObject();
             writer.WritePropertyName("file"u8);
-            writer.WriteBase64StringValue(Data.ToArray(), "D");
+#if NET6_0_OR_GREATER
+				writer.WriteRawValue(global::System.BinaryData.FromStream(Data));
+#else
+            using (JsonDocument document = JsonDocument.Parse(BinaryData.FromStream(Data)))
```

Comment:
Why convert a Stream to BinaryData before sending?  The purpose of taking a Stream in the public API is to prevent buffering it in memory, which is what BinaryData does.

Reply:
This is to persistent data. Current IPersistentModel has no stream API, so we need to convert it to BinaryData.

Reply:
there is a overload of JsonDocument.Parse(Value) that takes a `Stream`
maybe we could use this instead

Reply:
Why does this type need to implement `IJsonModel<T>`?  If the content in the stream is large, this could result in out of memory exceptions.  We should discuss whether there are any cases where this needs to happen, and evaluate the feature in the context of known use cases.  Without known use cases, the model should not implement this interface, because the negatives outweigh the positives.